### PR TITLE
Fixing #803

### PR DESCRIPTION
**Pull request by mpelikan**
Added support for binary numeric literal form.

### CODE REVIEW

File: js/lib/beautify.js [hunk 1]
```diff
@@ -1673,13 +1674,23 @@
...
+                    // switch to hex/oct/bin number, no decimal or e, just hex/oct/bin digits
                     allow_decimal = false;
                     allow_e = false;
                     c += input.charAt(parser_pos);
                     parser_pos += 1;
-                    local_digit = /[o]/.test(input.charAt(parser_pos)) ? digit_oct : digit_hex;
+                    if ( /[b]/.test(input.charAt(parser_pos)) ) {
+                        local_digit = digit_bin;
+                    }
+                    else {
```
Comment (bitwiseman) [inline js/lib/beautify.js]: Minor formatting and style changes:
```js
} else if ( /[o]/.test(input.charAt(parser_pos)) ) {
    local_digit = digit_oct;
} else {
    local_digit = digit_hex;
}
```

File: python/jsbeautifier/__init__.py [hunk 2]
```diff
@@ -1498,16 +1499,19 @@ def __tokenize_next(self):
...
                 allow_e = False
                 c += self.input[self.parser_pos]
                 self.parser_pos += 1
-                if re.match('[o]', self.input[self.parser_pos]):
-                    local_digit = self.digit_oct
+                if re.match('[b]', self.input[self.parser_pos]):
+                    local_digit = self.digit_bin
                 else:
-                    local_digit = self.digit_hex
+                    if re.match('[o]', self.input[self.parser_pos]):
```
Comment (bitwiseman) [inline python/jsbeautifier/__init__.py]: Same thing here as in js,  use `elif`.

**Comment by bitwiseman** [comment]
Style changes and tests needed. 
Thanks!

**Comment by mpelikan** [comment]
Tests already provided.

**Comment by bitwiseman** [comment]
For `0b` case?  I don't see them.

**Comment by mpelikan** [comment]
Oops. Sorry. I changed generated tests, not the source. Added test and fixed style to comply.

File: js/lib/beautify.js [hunk 3]
```diff
@@ -1673,13 +1674,21 @@
...
+                if (c === '0' && parser_pos < input_length && /[Xxob]/.test(input.charAt(parser_pos))) {
+                    // switch to hex/oct/bin number, no decimal or e, just hex/oct/bin digits
                     allow_decimal = false;
                     allow_e = false;
                     c += input.charAt(parser_pos);
                     parser_pos += 1;
-                    local_digit = /[o]/.test(input.charAt(parser_pos)) ? digit_oct : digit_hex;
+                    if ( /[b]/.test(input.charAt(parser_pos)) ) {
+                        local_digit = digit_bin;
+                    } else {
```
Comment (bitwiseman) [inline js/lib/beautify.js]: Use `} else if {`, then this will be good to merge.

Comment (mpelikan) [inline js/lib/beautify.js]: I disagree. That is not officially valid Javascript. There is no "else if" in the language. The "else" is always put in a else block. While the language will handle "else if", I am being explicit for clarity sake and to allow to extension of the else block.

This seems to be a nitpick for accepting a pull request.

Comment (bitwiseman) [inline js/lib/beautify.js]: This project uses `} else if (...) {` where reasonable and appropriate, this is one of those cases. 

We attempt to maintain parity between python and javascript versions of the code.  Not always possible, but where you see `elif ...:` in python, you should see `} else if (...) {` in javascript.

**Comment by bitwiseman** [comment]
Made the adjustment and committed.  Thanks for contributing!

**Comment by Nogrod** [comment]
breaks hex formatting -> example: 0xb401c4 will become 0x b401c4

**Pull request closed without merging**